Getting 404s when using via API

Following #8 .

Tried it. Got a 404 for files I was expecting it to serve.

Here is a PR branch where I expect it to work:
https://github.com/mobusoperandi/website/pull/111

There's a readme for this repo and a `default.nix` file, if you're a nix user.

`cargo run dev` is the command that is supposed to start the server. And then you should be able to ask for `/` or `/add.html`, for example and get a nice 200 but I get 404.

1. I didn't find the `default.nix` in the repository.
2. It gives many other errors when I ran 'cargo run dev':
```
    Finished dev [unoptimized + debuginfo] target(s) in 0.33s
     Running `target/debug/_ dev`
starting server at http://localhost:21928/
[Running 'cargo run -- build']
    Finished dev [unoptimized + debuginfo] target(s) in 0.33s
     Running `target/debug/_ build`
generating: "/add.html"
generating: "/favicon.ico"
generating: "/fullcalendar.js"
generating: "/index.html"
generating: "/inverticat.svg"
generating: "/mobs/agda_now.html"
generating: "/mobs/bass.html"
generating: "/mobs/liberty.html"
generating: "/mobs/malaga.html"
generating: "/mobs/mobrpg.html"
generating: "/mobs/more.html"
generating: "/mobs/pangolin.html"
generating: "/mobs/standard.html"
generating: "/twitter_logo.svg"
generating: "/vollkorn.ttf"
generating: "/youtube_logo.svg"
generating: "/zulip_logo.svg"
request https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg
request https://cdn.jsdelivr.net/npm/fullcalendar@6.0.2/index.global.min.js
request https://raw.githubusercontent.com/zulip/zulip/main/static/images/logo/zulip-icon-square.svg
request https://upload.wikimedia.org/wikipedia/commons/4/4f/Twitter-logo.svg
request https://upload.wikimedia.org/wikipedia/commons/0/09/YouTube_full-color_icon_%282017%29.svg
request https://gwfh.mranftl.com/api/fonts/vollkorn?download=zip&subsets=latin&variants=regular
response "HIT" https://raw.githubusercontent.com/zulip/zulip/main/static/images/logo/zulip-icon-square.svg
response "HIT" https://upload.wikimedia.org/wikipedia/commons/4/4f/Twitter-logo.svg
response "HIT" https://upload.wikimedia.org/wikipedia/commons/0/09/YouTube_full-color_icon_%282017%29.svg
response "HIT" https://gwfh.mranftl.com/api/fonts/vollkorn?download=zip&subsets=latin&variants=regular
response "HIT" https://cdn.jsdelivr.net/npm/fullcalendar@6.0.2/index.global.min.js
response "MISS" https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg
node:internal/modules/cjs/loader:1060
  const err = new Error(message);
              ^

Error: Cannot find module '@tailwindcss/typography'
Require stack:
- /home/lomirus/Projects/GITHUB/website/tailwind.config.js
- /home/lomirus/.npm/_npx/bc7494899bbce1ca/node_modules/tailwindcss/lib/cli.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1060:15)
    at Module._load (node:internal/modules/cjs/loader:905:27)
    at Module.require (node:internal/modules/cjs/loader:1127:19)
    at require (node:internal/modules/helpers:112:18)
    at Object.<anonymous> (/home/lomirus/Projects/GITHUB/website/tailwind.config.js:3:5)
    at Module._compile (node:internal/modules/cjs/loader:1246:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1300:10)
    at Module.load (node:internal/modules/cjs/loader:1103:32)
    at Module._load (node:internal/modules/cjs/loader:942:12)
    at Module.require (node:internal/modules/cjs/loader:1127:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/lomirus/Projects/GITHUB/website/tailwind.config.js',
    '/home/lomirus/.npm/_npx/bc7494899bbce1ca/node_modules/tailwindcss/lib/cli.js'
  ]
}

Node.js v19.6.0
thread 'main' panicked at 'assertion failed: output.status.success()', src/tailwind.rs:37:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Finished running. Exit status: 101]
```

Can you provide a minimal reproducible example?

--- a/src/server.rs
+++ b/src/server.rs
@@ -96,7 +96,9 @@ async fn static_assets(
     root: PathBuf,
 ) -> Result<Response, tide::Error> {
     // Get the path and mime of the static file.
-    let mut path = root.join(req.url().path());
+    let mut path = req.url().path().to_string();
+    path.remove(0);
+    let mut path = root.join(path);
     if path.is_dir().await {
         path.push("index.html");
     }

